base/abstractarraymath.jl

```diff
@@ -166,15 +166,7 @@ end
 
 ## Permute array dims ##
 
-function permutedims(B::AbstractArray, perm)
-    dimsB = size(B)
-    ndimsB = length(dimsB)
-    (ndimsB == length(perm) && isperm(perm)) || throw(ArgumentError("no valid permutation of dimensions"))
-    dimsP = ntuple(i->dimsB[perm[i]], ndimsB)::typeof(dimsB)
-    P = similar(B, dimsP)
-    permutedims!(P, B, perm)
-end
-
+permutedims(B::AbstractArray, perm) = permutedims(convert(Array, B), perm)
 
```

Comment:
perhaps not full then, but if it's avoiding methoderrors that you're worried about, converting to Array can also be done at the call site. I see Images.jl has a definition like this that extends permutedims for AbstractArray, that seems bad to change Base behavior on types not owned by the package.